common/src/main/java/io/netty/util/DomainNameMapping.java

```diff
@@ -72,39 +75,32 @@ public DomainNameMapping(int initialCapacity, V defaultValue) {
      * </p>
      *
      * @param hostname the host name (optionally wildcard)
-     * @param output the output value that will be returned by {@link #map(String)} when the specified host name
-     *               matches the specified input host name
+     * @param output   the output value that will be returned by {@link #map(String)} when the specified host name
+     *                 matches the specified input host name
+     * @deprecated use {@link DomainMappingBuilder} to create and fill the mapping instead
      */
+    @Deprecated
     public DomainNameMapping<V> add(String hostname, V output) {
-        if (hostname == null) {
-            throw new NullPointerException("input");
-        }
-
-        if (output == null) {
-            throw new NullPointerException("output");
-        }
-
-        map.put(normalizeHostname(hostname), output);
+        map.put(
+            normalizeHostname(checkNotNull(hostname, "hostname")),
+            checkNotNull(output, "output"));
         return this;
     }
 
     /**
      * Simple function to match <a href="http://en.wikipedia.org/wiki/Wildcard_DNS_record">DNS wildcard</a>.
      */
-    private static boolean matches(String hostNameTemplate, String hostName) {
+    static boolean matches(String hostNameTemplate, String hostName) {
         // note that inputs are converted and lowercased already
-        if (DNS_WILDCARD_PATTERN.matcher(hostNameTemplate).matches()) {
-            return hostNameTemplate.substring(2).equals(hostName) ||
-                    hostName.endsWith(hostNameTemplate.substring(1));
-        } else {
-            return hostNameTemplate.equals(hostName);
-        }
+        return hostNameTemplate.startsWith("*.")
+            ? hostNameTemplate.substring(2).equals(hostName) || hostName.endsWith(hostNameTemplate.substring(1))
```

Comment:
can we avoid using `substring` and creating two new objects? Consider using `regionMatches` instead.